### ISSUE TITLE
Sprint 1.32.0 2/2 - Jan 10 to Jan 21

### ISSUE THREAD

**Issue by marcushyett-ph** [PostHog/posthog#7893]
# Global Sprint Planning

## Retro: Status of Outcomes from Previous Sprint
If you're a goal owner please update status with: (failed it / nailed it / scraped it) and any brief additional context.

1. **Plugin Server Scalability:** Fix overloading issues on plugin server worker threads @yakkomajuri 
   **Status**: @yakkomajuri  Not Done.
2. **Nail Deployments:** CLI to make helm chart deploys seamless @fuziontech 
    **Status**: @fuziontech  Depriorized for Async Migrations, still important but not for next sprint
3. **Performance:** Optimize most query performance large customers to <5seconds @macobo 
   - **Status:** Not complete, migration will complete soon on cloud - queries will get faster after.
      - Done: [person_distinct_id table being slow >10M users](https://github.com/PostHog/posthog/issues/7663). Deployed a new table for all users on cloud (and migration for self-hosted). Took many queries from 20s -> 6s.
      - Not done: [events table has a bad sort key](https://github.com/PostHog/posthog/issues/5684) - #team-platform started a migration on Dec 21st and expected it to take 4-5 days. Still ongoing at 17 days. Fixing this will speed up trends, funnels and retention queries significantly.
      - Smaller issues tackled: 
          - Mapped out slow key client queries, found sticking points.
          - We materialized all of our key clients properties, this sped up most of their insights using properties. Blocked by events table migration to further improve this
          - Removed minute interval + improved large time range handling: https://github.com/PostHog/posthog/issues/7625 https://github.com/PostHog/posthog/issues/7626
          - Porting lifecycle query - ongoing https://github.com/PostHog/posthog/issues/7382
          - Benchmarking switching join order and alternate storage mechanisms: https://github.com/PostHog/posthog/issues/7874
      - Also there's now a RFC with a plan for achieving performance greatness: https://github.com/PostHog/product-internal/pull/240
      - Interviews planned to validate some ambitious performance-related proposals
4. **Experimentation:** Users can run multiple multivariate experiments & take decisions based on the results of those experiments @neilkakkar 
    **Status**: Not complete
    - Had issues scheduling interviews
    - switched track a bit to including other core features: Trends metrics + multivariate experiments
    - Result page design & clarifications pushed to next sprint
    - Focus on gathering feedback next sprint + polish 
5. **Nail Insights date support** @pauldambra 
    **Status**: 
   - in progress
   - ability to query any property as if its a date released behind a flag
   - added new fields to property definition
   - next step is to start auto-detecting dates and times
6. **Make Dashboards great again** @Twixes 
    **Status**: @Twixes  please update
7.  **Collaboration: Taxonomy:** Users should understand events actions and properties others have set up @alexkim205 
    **Status**: @alexkim205 please update

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer.

1. “Get 5 users running real experiments and sharing their experience” @neilkakkar 
2. Close out diagnosing causes and groups @EDsCODE 
3. Insights cleanup and scalability @macobo 
4. Make dashboards great again @Twixes 
5. **Collaboration: Taxonomy:** Users should understand events actions and properties others have set up @alexkim205
6.  Converge Helm chart and cloud: Support roll-forward and back for helm @fuziontech 
7.  Nail async migrations @tiina303 

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Low priority / side quests

-

```

**Comment by paolodamico** [comment]
# WIP - Core Experience

Support Hero: @Twixes (W1)

- Make dashboards great (@Twixes)
- Users should understand events, actions and properties others have set up (@alexkim205)
- Date/time support (part of taxonomy, part of collaboration) (@pauldambra)

------
- Implement enterprise permissions (*)

Team Core Experience

## Retro
- Dashboards - @Twixes 
  - Quite a bit of work pulled into the next sprint. Uncertainty on the final design and UX. 
  - Bit lag on feedback due to vacations and timezones.
  - Hackathon   
 - @alexkim205 Could've done a better job involving the team sooner on tooltip questions - could've implemented a stop gap solution to get it shipped sooner. 
- @pauldambra - "kids and life™". Pleased with datetime stuff - backend query work was quick. Should be repeatable work for other queries. (each insight, cohort, recording, etc needs updating)

## Hang over items from previous sprint
- Dashboards - shipping a new insight card component
    - Under a feature flag, incomplete but most of the way there
- Tooltips - documentation to capture state of work

## Planning
### High priority
- @alexkim205 - Taxonomy - https://github.com/PostHog/posthog/issues/7589
    - Events tab redesign is taking more of the lead on UX changes
    - Stats and definitions easier to discover, navigate, etc. 
    - Filter building - yet to be solidified. Needs UX guidance. 
    - Backend task - how we tag and identify different kinds of events (bad events, data types, timestamps etc) - Classification and stats that are queryable. 
@pauldambra - Get auto-detection in for datetime for customer feedback. Work through insight types. 
- Need to think about manual correction. For example they have a datetime string and they want to use it as a string. Would sit only in the paid version for taxonomy management. 

### Low priority / side quests
- @Twixes - deploying demo.posthog.com

**Comment by marcushyett-ph** [comment]
@fuziontech @EDsCODE Can you add your retro sprint plans go for this sprint?

**Comment by EDsCODE** [comment]
## Retro

- @macobo lot of planning for optimization improvements and benchmarking done
- @neilkakkar experiments feedback and continued improvement. Performance improvements
- @liyiy Experiments frontend
- @EDsCODE Clearing code, got the backend for insight cohorting fully merged. Made progress on closing out diagnosing causes (below)
- @hazzadous clickhouse bug squashing, lifecycle query refactor

## Planning

**Lifecycle wrapup** - @macobo 
- Supporting person property pushdowns
- Supporting materialized columns
- Created_at usage (done?)
- Stretch: Groups support

**Experimentation** - @neilkakkar AND @liyiy 
- Get 5 users running real experiments and sharing their experience
- Polish end to end flow: from creating experiment to extracting insights from it
- Allow having secondary metrics to track

**Closing out diagnosing causes** - @EDsCODE 
- Expose window id, session id, timestamp on funnel and path queries

**Closing out Groups** - @macobo 
- Deleting a group 
- Renaming group type https://github.com/PostHog/posthog/issues/7763
- Show related group names https://github.com/PostHog/posthog/issues/7940
- Side quest: Prepare a overarching proposal for persons/groups search and cohort creation: https://github.com/PostHog/posthog/issues/7941

**Insight Cleanup:** - @EDsCODE 
- Has done X in time range (on insights) https://github.com/PostHog/posthog/issues/5854
- OR Filtering (on insights) https://github.com/PostHog/posthog/issues/5854
- Increase time range flexibility on action based cohorts
- Insight cohorting
- Benchmarking high scale improvement possibilities
- Clearing out postgres specific code https://github.com/PostHog/posthog/pull/7830

**Comment by fuziontech** [comment]
## Platform Retro 👾

- @fuziontech - Vacation 🏝 
- @tiina303 - Vacation 🏝
- @yakkomajuri - Vacation 🏝
- @guidoiaquinti - Vacation 🏝 and 😷
- @hazzadous - First Sprint w/ Platform 🍻🎉
- @marcushyett-ph - Shipped 2 ridiculously valuable plugins 🚢

## Planning

**Plugins (late addition P0)** - @yakkomajuri & @tiina303 
- Debug why plugins stop occasionally 
- Increase default number of partitions for helm chart https://github.com/PostHog/charts-clickhouse/pull/243
- Investigate PG timeouts from Plugin server
- Health check endpoint for Plugins server 
- Last event seen at surfaced in instance status dashboard
- Identify other means to improve performance and reliability

**Async Migrations** - @yakkomajuri & @tiina303
- Reliability improvements - specifically around making sure all migrations are idempotent and resumable
- Look into other alternate ways of doing async migrations (split out as service or use off the shelf tooling) for long term
- (Other Improvements)[https://github.com/PostHog/product-internal/pull/243]

**Helm Chart** - @guidoiaquinti & @hazzadous
- Hooks -> resources on helm chart to minimize downtime on deploys and upgrades https://github.com/PostHog/charts-clickhouse/pull/179
- Better support for external resources on helm chart for testing and larger deployments https://github.com/PostHog/charts-clickhouse/pull/240
- Get our helm chart into a state that matches ECS robustness of deploys

**Carry over / Misc**
- Finish out of order event processing - @tiina303
- [K8s Eng hiring](https://posthog.workable.com/backend/jobs/2105790/browser/applied) - @fuziontech
- Platform Team Rechartering
- https://github.com/PostHog/posthog/issues/7971 - @fuziontech
